article/app/renderers/RemoteRenderer.scala

```diff
@@ -24,36 +26,59 @@ class RemoteRenderer {
 
   private[this] val SCHEMA = "schema/dotcomponentsDataModelV2.jsonschema"
 
-  private[this] def validate(model: DotcomponentsDataModel) = {
+  private[this] val circuitBreaker = CircuitBreakerRegistry.withConfig(
+    name = "dotcom-rendering-client",
+    system = ActorSystem("dotcom-rendering-client-circuit-breaker"),
+    maxFailures = Configuration.rendering.circuitBreakerMaxFailures,
+    callTimeout = Configuration.rendering.timeout.plus(200.millis),
+    resetTimeout = Configuration.rendering.timeout * 4,
+  )
+
+  private[this] def validate(model: DotcomponentsDataModel): JsResult[JsValue] = {
     val rawschema: String = Source.fromResource(SCHEMA).getLines.mkString("\n")
     val schema = Json.fromJson[SchemaType](Json.parse(rawschema)).get
     val validator = new SchemaValidator(Some(Version7))(Lang.Default)
+
     validator.validate(schema, DotcomponentsDataModel.toJson(model))
   }
 
   private[this] def get(
-    ws:WSClient,
+    ws: WSClient,
     payload: String,
     article: PageWithStoryPackage,
     endpoint: String
   )(implicit request: RequestHeader): Future[Result] = {
 
-    ws.url(endpoint)
-      .withRequestTimeout(2000.millis)
-      .addHttpHeaders("Content-Type" -> "application/json")
-      .post(payload)
-      .map(response => {
-        response.status match {
-          case 200 =>
-            Cached(article)(RevalidatableResult.OkDotcomponents(Html(response.body)))
-          case _ =>
-            throw new Exception(response.body)
-        }
-      })
+    def get(): Future[Result] = {
+      ws.url(endpoint)
+        .withRequestTimeout(Configuration.rendering.timeout)
+        .addHttpHeaders("Content-Type" -> "application/json")
+        .post(payload)
+        .map(response => {
+          response.status match {
+            case 200 =>
+              Cached(article)(RevalidatableResult.OkDotcomponents(Html(response.body)))
+            case _ =>
+              throw new Exception(response.body)
+          }
+        })
+    }
+
+
+    if (CircuitBreakerSwitch.isSwitchedOn) {
+      circuitBreaker.withCircuitBreaker(get())
```

Comment:
What is the application behaviour when the circuit breaker is closed and it gets a request, does it generate an exception resulting in a 500? Block the thread? etc

Reply:
It returns a Future failed with a scala.concurrent.TimeoutException. But it's a good point as it doesn't look like we handle failed futures at all for DCR. I'll investigate more, but we should add least handle these and return a 5xx.

Reply:
I think it'll most likely bubble the exception up and cause a 500 anyway so that's probably fine